frontend/src/components/Modal/DeleteRecurRideModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+import Modal from "./Modal";
+import { Ride } from '../../types/index';
+import { Button, Input, Label } from '../../components/FormElements/FormElements';
+import styles from './deleteModal.module.css';
+
+type DeleteSingleRideModalProps = {
+    open: boolean,
+    ride: Ride,
+}
+
+const DeleteRecurRideModal = ({ open, ride }: DeleteSingleRideModalProps) => {
+    const [isOpen, setIsOpen] = useState(open);
+    const [cancelSingle, setCancelSingle] = useState(false);
+
+    const changeSelection = (e: any) => {
+        if (e.target.value === 'single') setCancelSingle(true);
+        else setCancelSingle(false);
+    }
+
+    const closeModal = () => {
+        setIsOpen(false);
+    }
+
+    // TODO
+    const confirmCancel = () => {
+        console.log('deleted')
+    }
```

Comment:
for single ride only, you can reference the endpoint in #176, ill try to add documentation in the api doc soon. for all recurring rides, you can use the normal delete endpoint.